tests/pytests/unit/engines/test_libvirt_events.py

```diff
@@ -0,0 +1,202 @@
+"""
+unit tests for the libvirt_events engine
+"""
+import pytest
+import salt.engines.libvirt_events as libvirt_events
+from tests.support.mock import MagicMock, patch
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {libvirt_events: {}}
+
+
+@pytest.fixture
+def mock_libvirt():
+    try:
+        patcher = patch("salt.engines.libvirt_events.libvirt")
+        mock_libvirt = patcher.start()
+        mock_libvirt.getVersion.return_value = 2000000
+        mock_libvirt.virEventRunDefaultImpl.return_value = -1  # Don't loop for ever
+        mock_libvirt.VIR_DOMAIN_EVENT_ID_LIFECYCLE = 0
+        mock_libvirt.VIR_DOMAIN_EVENT_ID_REBOOT = 1
+        mock_libvirt.VIR_STORAGE_POOL_EVENT_ID_LIFECYCLE = 0
+        mock_libvirt.VIR_STORAGE_POOL_EVENT_ID_REFRESH = 1
+        mock_libvirt.VIR_NODE_DEVICE_EVENT_ID_LIFECYCLE = 0
+        mock_libvirt.VIR_NODE_DEVICE_EVENT_ID_UPDATE = 1
+        yield mock_libvirt
+    finally:
+        patcher.stop()
+
+
+@patch(
+    "salt.engines.libvirt_events.libvirt",
+    VIR_PREFIX_NONE=0,
+    VIR_PREFIX_ONE=1,
+    VIR_PREFIX_TWO=2,
+    VIR_PREFIX_SUB_FOO=0,
+    VIR_PREFIX_SUB_BAR=1,
+    VIR_PREFIX_SUB_FOOBAR=2,
+)
+def test_get_libvirt_enum_string_subprefix(libvirt_mock):
+    """
+    Make sure the libvirt enum value to string works reliably with
+    elements with a sub prefix, eg VIR_PREFIX_SUB_* in this case.
+    """
+    # Test case with a sub prefix
+
+    assert libvirt_events._get_libvirt_enum_string("VIR_PREFIX_", 2) == "two"
+
+
+@patch("salt.engines.libvirt_events.libvirt", VIR_PREFIX_FOO=0, VIR_PREFIX_BAR_FOO=1)
```

Comment:
Same as above.
Of you end up needing to patch the same thing over and over, use a fixture. You can then decorate the test function with `@pytest.mark.usefixture("the fixture name")`